.github/workflows/iroha2-dev.yml

```diff
@@ -176,33 +176,27 @@ jobs:
 
   # Coverage is both in PR and in push pipelines so that:
   # 1. PR can get coverage report from bot.
-  # 2. Coverage bot can have results from
-  # `iroha2-dev` to report coverage changes.
+  # 2. Coverage bot can have results from `iroha2-dev` to report coverage changes.
   coverage:
     runs-on: ubuntu-latest
     container:
       image: 7272721/i2-ci:latest
     steps:
-      # TODO Remove this step #2165
-      - name: Add component
-        run: rustup component add llvm-tools-preview
-
+      # TODO Remove these steps #2165
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: llvm-tools-preview
+      - uses: taiki-e/install-action@cargo-llvm-cov
+      #
       - uses: actions/checkout@v2
       - uses: Swatinem/rust-cache@v1
-      - name: Run tests
-        run: |
-          mold --run cargo test \
-          --quiet --workspace --no-fail-fast -- \
-          --skip unstable_network --skip ui --test-threads 3
-        env:
-          RUSTFLAGS: "-C instrument-coverage"
-          LLVM_PROFILE_FILE: "iroha-%p-%m.profraw"
-      - name: Generate a grcov coverage report
-        run: |
-          grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "/*" -o lcov.info
+      - name: Run tests and generate lcov report
         if: always()
+        run: mold --run cargo llvm-cov --all-features --workspace --lcov --output-path lcov.info
       - name: Upload coverage to codecov.io
         uses: codecov/codecov-action@v2
-        if: always()
         with:
           file: lcov.info
+          fail_ci_if_error: true
```

Comment:
This is dangerous. We have flaky tests so we wouldn't be able to push anything if the coverage fails.

Reply:
If it enters this uploading step, that means all the tests have passed in the previous step.
But I agree to remove this line because we would not like to withhold pushing while codecov.io is down

Reply:
But in that case we could dismiss the failing check and merge

Reply:
I think it affects only this `coverage` job, but not clear. I'll remove this line